### PR TITLE
moving all of resources to ChefResource class under libraries

### PR DESCRIPTION
**Pull request by Anirudh-Gupta**
It's good to encapsulate all of our  resources under Chef::Resource class so that the cookbook users can easily add it as a dependency to their cookbooks and can add their own resources(built on top of resources shipped with this cookbook) under Resources folder. Improves readability as well.

### CODE REVIEW

**Comment by bikbajwa** [comment]
Hey Anirudh, I think keeping the resources under the resources folder is the generally accepted way of how to do custom resources in Chef. Is there a doc that you are referencing for this change?

**Comment by Anirudh-Gupta** [comment]
Let's say for example, If my cookbook has a dependency on this ilo-chef cookbook and I feel I want to dump more of the computer details leveraging the existing dump action that has been written  in the ilo_computer_details resource, I can easily extend this class if we have this action already encapsulated in a class. I can monkey patch it. On the contrary, it would be difficult(not saying, its not possible. Its just not much intuitive) for me to extend this action if I find this action in the resource.
For a reference, have a look at this cookbook by Chef containing custom Docker resources - https://github.com/chef-cookbooks/docker
or 
this one - https://github.com/chef-cookbooks/audit
More generally accepted way is putting it into the Chef::Resource class.

**Comment by bikbajwa** [comment]
Okay, I understand what you are saying now and I agree. It's a :+1:  from me. @jsmartt any thoughts?

**Comment by jsmartt** [comment]
It's a little hard to see the code changes, so it may take some time to review. I do like how the docker cookbook used a [base class](https://github.com/chef-cookbooks/docker/blob/master/libraries/docker_base.rb) to inherit from, so each resource class wouldn't have to import all the helpers and do all the basic setup (ie require, import, and include the helper class, as well as setting the `ilos` property).

Also, just after first glance, I see we're doing a `include IloCookbook::IloHelper` in each resource, but we're also doing 
```ruby
action_class do
  include IloHelper
end
```
in each class too. Is this a necessary duplication?

File: libraries/ilo_helper.rb [hunk 1]
```diff
@@ -10,33 +10,36 @@
...
-    require 'ilo-sdk'
-    Chef::Log.debug("Found gem ilo-sdk (version #{node['ilo']['ruby_sdk_version']})")
-  rescue LoadError
-    Chef::Log.info("Did not find gem ilo-sdk (version #{node['ilo']['ruby_sdk_version']}). Installing now")
-    chef_gem 'ilo-sdk' do
-      version node['ilo']['ruby_sdk_version']
-      compile_time true if Chef::Resource::ChefGem.method_defined?(:compile_time)
+module IloCookbook
+  ## Helper module
+  module IloHelper
```
Comment (jsmartt) [inline libraries/ilo_helper.rb]: Also, since this is now scoped inside the `IloCookbook` module, we can probably rename it to just `Helper`

**Comment by Anirudh-Gupta** [comment]
@jsmartt - It was an unnecessary duplication. Please review it.

**Comment by jsmartt** [comment]
Re-opening in #6

**Pull request closed without merging**